### ISSUE TITLE
Ruby 3 Upgrade

### ISSUE THREAD

**Issue by joncameron** [avalonmediasystem/avalon#5008]
# Description
Ruby 2.x will reach end of life soon. Avalon needs to move to Ruby 3.1 (or latest stable 3.x) to keep current with the supported releases of Ruby. 

# Done Criteria
- [ ] Dependencies are updated to Ruby 3 and are available 
- [ ] Upgrade Avalon and evaluate (what breaks?)

# Questions
- End of March-ish will be the Ruby 2.7 EOL. Should we get this into a release before then? (Current status color: yellow)
- Current Samvera dependencies (core components) are using Ruby 2.7; full move to Ruby 3 will requires updates to these core components as well (ex: Active Fedora)
- Past Active Fedora, it shouldn't be extensive to upgrade the rest of the necessary dependencies to Ruby 3
- Assume for this issue that dependency updates are already done

**Comment by cjcolvar** [comment]
I believe the only Samvera dependencies we're waiting on are:
- [ ] active_fedora
- [ ] hydra-head
- [ ] noid-rails (This has a PR waiting to be merged which adds support)

**Comment by elynema** [comment]
Tests are now running. Waiting on tamsin for the active-triples gem cut, which will be after the new year.

**Comment by cjcolvar** [comment]
One failing test due to a bug in edtf-ruby that needs to be merged and released: https://github.com/inukshuk/edtf-ruby/compare/main...cjcolvar:edtf-ruby:patch-1

**Comment by elynema** [comment]
Have a draft pull request - just waiting on Samvera gem releases.
